cpp/src/io/parquet/writer_impl.cu

```diff
@@ -85,6 +87,49 @@ parquet::Compression to_parquet_compression(compression_type compression)
   }
 }
 
+size_type column_size(column_view const& column, rmm::cuda_stream_view stream)
+{
+  if (column.num_children() == 0) {
+    return is_fixed_width(column.type()) ? size_of(column.type()) * column.size() : 0;
+  }
```

Comment:
here as well, should this throw if we hit a non-fixed width type with no children? If we throw at the bottom, this can become
```suggestion
  if (column.num_children() == 0 and is_fixed_width(column.type())) {
    return size_of(column.type()) * column.size();
  }
```

Reply:
I'll have to check again on this one, but I think I did it this way because of some of the empty table tests.

Reply:
looks good!